File: frontend/app/domain/dto/getUserInfoResponse.ts
import { UserProfile } from './userProfile';
import { UserBadge } from './userBadge';
import { MenuEntryFavorite } from './menuEntryFavorite';
import { LastOrder } from './lastOrder';

export class GetUserInfoResponse implements app.domain.dto.IGetUserInfoResponse, Serializable<GetUserInfoResponse> {
    constructor() { }

    id: string;
    userName: string;
    balance: number;
    profile: UserProfile;
    badges: UserBadge[];
    favorites: MenuEntryFavorite[];
    userToken: string;
    last5Orders: LastOrder[];
    roles: string[];

    deserialize(input: any): GetUserInfoResponse {
        this.id = input.id;
        this.userName = input.userName;
        this.balance = input.balance;
        this.profile = new UserProfile().deserialize(input.profile);
        this.badges = new Array<UserBadge>();

        if (input.badges) {
            for (var badge of input.badges) {
                this.badges.push(new UserBadge().deserialize(badge));
            }
        }
        this.favorites = new Array<MenuEntryFavorite>();
        if (input.favorites) {
            for (var favorite of input.favorites) {
                this.favorites.push(new MenuEntryFavorite().deserialize(favorite));
            }
        }
        this.userToken = input.userToken;
        this.last5Orders = new Array<LastOrder>();
        if (input.last5Orders) {
            for (var last5Order of input.last5Orders) {
                this.last5Orders.push(new LastOrder().deserialize(last5Order));
            }
        }
        this.roles = input.roles;

        return this;
    }
}

File: backend/WebApi/Lunchorder.Domain/Dtos/Responses/GetUserInfoResponse.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;
using Newtonsoft.Json.Serialization;

namespace Lunchorder.Domain.Dtos.Responses
{
    public class GetUserInfoResponse
    {
        public Guid Id { get; set; }
        public string UserName { get; set; }
        public decimal Balance { get; set; }
        public UserProfile Profile { get; set; }
        public IEnumerable<UserBadge> Badges { get; set; }
        public IEnumerable<MenuEntryFavorite> Favorites { get; set; }
        public IEnumerable<LastOrder> Last5Orders { get; set; }
        public IEnumerable<UserBalanceAuditItem> Last5BalanceAuditItems { get; set; }
        public IEnumerable<string> Roles { get; set; }

        /// <summary>
        /// A new token for the user in case of user creation in our database.
        /// </summary>
        public string UserToken { get; set; }

        public string ToJson()
        {
            //Return json
            return JsonConvert.SerializeObject(this, new JsonSerializerSettings
            {
                ContractResolver = new CamelCasePropertyNamesContractResolver()
            });
        }
    }
}
